crates/matrix-sdk/src/authentication/oauth/qrcode/reciprocate/scanned.rs

```diff
@@ -0,0 +1,728 @@
+// Copyright 2025 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The reciprocation flow where the existing device generates a QR code to be
+//! scanned by the new device.
+
+use eyeball::SharedObservable;
+use futures_core::Stream;
+use matrix_sdk_base::crypto::types::{SecretsBundle, qr_login::QrCodeData};
+use url::Url;
+
+use super::{
+    AlmostEstablishedSecureChannel, Error, SecureChannel, export_secrets_bundle, finish_login_grant,
+};
+use crate::Client;
+
+/// A handler for reciprocating an existing login to a new device.
+pub struct ReciprocateHandler {
+    /// The existing client.
+    client: Client,
+    /// The secrets that will be transmitted to the new device.
+    secrets_bundle: SecretsBundle,
+    /// The secure channel used to communicate with the new device.
+    channel: Channel,
+    /// The QR code data to be scanned by the new device.
+    qr_code_data: QrCodeData,
+    /// The current state of the reciprocation process.
+    state: SharedObservable<ReciprocateProgress>,
+}
+
+impl std::fmt::Debug for ReciprocateHandler {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("ReciprocateHandler")
+            .field("secrets_bundle", &"SecretsBundle")
+            .field("channel", &"Channel")
```

Comment:
Is there a reason to include those two? Why not remove the fields and just use `finish_non_exhaustive()` to indicate that not all fields are printed.